### PR TITLE
Add BremsstrahlungProcess

### PR DESCRIPTION
**Pull request by vrpascuzzi**
Added bremsstrahlung process to LDemoParams, and
ImportedProcesses and  Seltzer-Berger tests. For
consistency, the following additional modifications
to ImportProcess were made:

  * `mu_brems` -> `mu_brem`
  * `h_brems` -> `h_brem`

### CODE REVIEW

**Comment by sethrj** [comment]
Unfortunately the "build" output for jenkins is pretty hard to parse. Most of the warnings are the usual vecgeom recursion crap; the first build error is just with the CI machine again:
```
fatal error: error writing to /tmp/cc0VuVGc.s: No space left on device
```
and then another build error:
```
./app/geant-exporter/ImportProcessConverter.cc: In function 'celeritas::ImportProcessClass geant_exporter::{anonymous}::to_import_process_class(const G4VProcess&)':
../app/geant-exporter/ImportProcessConverter.cc:101:52: error: 'h_brems' is not a member of 'celeritas::ImportProcessClass'; did you mean 'h_brem'?
  101 |             {"hBrems",         ImportProcessClass::h_brems},
      |                                                    ^~~~~~~
      |                                                    h_brem
../app/geant-exporter/ImportProcessConverter.cc:112:52: error: 'mu_brems' is not a member of 'celeritas::ImportProcessClass'; did you mean 'mu_brem'?
  112 |             {"muBrems",        ImportProcessClass::mu_brems},
      |                                                    ^~~~~~~~
      |                                                    mu_brem
```
(you might have to expand the full log to see these)

**Comment by vrpascuzzi** [comment]
Thanks, @sethrj! Now I know to expand the full log.

**Comment by vrpascuzzi** [comment]
Looks like we'll need some intervention for the CI.

File: test/physics/em/ImportedProcesses.test.cc [hunk 1]
```diff
@@ -196,6 +197,40 @@ TEST_F(ImportedProcessesTest, photoelectric)
...
+    ASSERT_EQ(1, all_applic.size());
+    Applicability applic = *all_applic.begin();
+
+    // Test step limits
+    for (auto mat_id : range(MaterialId{materials_->num_materials()}))
+    {
+        applic.material = mat_id;
+        auto builders   = process->step_limits(applic);
+        EXPECT_TRUE(builders[VGT::macro_xs]);
+        EXPECT_FALSE(builders[VGT::energy_loss]);
```
Comment (sethrj) [inline test/physics/em/ImportedProcesses.test.cc:229]: I thought brems has a de/dx component for electrons below the production threshold?

File: app/geant-exporter/ImportProcessConverter.cc [hunk 2]
```diff
@@ -98,7 +98,7 @@ ImportProcessClass to_import_process_class(const G4VProcess& process)
             {"ionIoni",        ImportProcessClass::ion_ioni},
             {"msc",            ImportProcessClass::msc},
             {"hIoni",          ImportProcessClass::h_ioni},
-            {"hBrems",         ImportProcessClass::h_brems},
+            {"hBrem",          ImportProcessClass::h_brem},
```
[the remark this answers is not in the document]
Comment (stognini) [reply app/geant-exporter/ImportProcessConverter.cc:101]: Yes. It is `hBrems` in Geant4:

`G4hBremsstrahlung.hh`
```c++
  explicit G4hBremsstrahlung(const G4String& processName = "hBrems");
```

File: test/physics/em/ImportedProcesses.test.cc [hunk 3]
```diff
@@ -196,6 +197,40 @@ TEST_F(ImportedProcessesTest, photoelectric)
...
+        SKIP("Failed to create process: " << e.what());
+    }
+
+    // Test model
+    auto models = process->build_models(ModelIdGenerator{});
+    ASSERT_EQ(1, models.size());
+    ASSERT_TRUE(models.front());
+    EXPECT_EQ("Seltzer-Berger bremsstrahlung", models.front()->label());
+    auto all_applic = models.front()->applicability();
+    ASSERT_EQ(1, all_applic.size());
```
Comment (amandalund) [inline test/physics/em/ImportedProcesses.test.cc:220]: I think the size should be 2 (both electron and positron).

File: app/geant-exporter/ImportProcessConverter.cc (hunk 2 above)
[the remark this answers is not in the document]
Comment (sethrj) [reply app/geant-exporter/ImportProcessConverter.cc:101]: Thanks!

File: test/physics/em/ImportedProcesses.test.cc (hunk 3 above)
Comment (sethrj) [reply test/physics/em/ImportedProcesses.test.cc:220]: I have a feeling this test doesn't pass yet ;)

Comment (vrpascuzzi) [reply test/physics/em/ImportedProcesses.test.cc:220]: Right. I tried running this test but apparently didn't configure my celeritas build to use ROOT so it doesn't run....

File: test/physics/em/ImportedProcesses.test.cc (hunk 1 above)
Comment (vrpascuzzi) [reply test/physics/em/ImportedProcesses.test.cc:229]: This is true.

File: app/geant-exporter/ImportProcessConverter.cc (hunk 2 above)
[the remark this answers is not in the document]
Comment (vrpascuzzi) [reply app/geant-exporter/ImportProcessConverter.cc]: Same for `muBrems`.

[the remark this answers is not in the document]
Comment (sethrj) [reply app/geant-exporter/ImportProcessConverter.cc]: @vrpascuzzi it looks like your PR still changes the enum, even though it was originally "correct" (written as intended for consistency with the G4 class names). I see your note that you changed these for "consistency", but I don't understand with what?

[the remark this answers is not in the document]
Comment (vrpascuzzi) [reply app/geant-exporter/ImportProcessConverter.cc]: There is a mix of `*brem` and `*brems` in these import classes, e.g., [`h_brems`](https://github.com/celeritas-project/celeritas/blob/master/app/geant-exporter/ImportProcessConverter.cc#L101), [`e_brem`](https://github.com/celeritas-project/celeritas/blob/master/app/geant-exporter/ImportProcessConverter.cc#L105), [`mu_brems`](https://github.com/celeritas-project/celeritas/blob/master/app/geant-exporter/ImportProcessConverter.cc#L112).

Looking again at @stognini's reference and related ones, I understand now the reason for doing it this way. However, I think within celeritas it's important we have a common 'value' that maps to whatever mindset G4 had when aliasing their processes.

What I'm suggesting is that, in `ImportProcess.hh`, we use `h_brems`, `e_brems` and `mu_brems` rather than [...], `e_brem` [...]. It's OK to follow G4 to some extent but also must keep our sanity (and spare that of future users) when we come across such apparent illogical conventions.

So we can either (a) revert these changes or (b) make things right. :)

[the remark this answers is not in the document]
Comment (sethrj) [reply app/geant-exporter/ImportProcessConverter.cc]: Aha, I see. Well, you know I'm a stickler for consistency and not often a fan of "this is how Geant does it". So I'm up for changing! Since I thought the natural abbreviation (at least how I speak and perceive others shortcuts as well) was "brems" I'm making a poll 😉 

https://celeritasproject.slack.com/archives/CSLM7G17V/p1622288259002800

[the remark this answers is not in the document]
Comment (sethrj) [reply app/geant-exporter/ImportProcessConverter.cc]: And now that we're in the consistency can of worms... Geant says `Rayl` and `compt` as if Rayleigh is a person and Compton isn't 🙄

File: app/geant-exporter/ImportProcessConverter.cc [hunk 4]
```diff
@@ -136,7 +136,7 @@ ImportModelClass to_import_model(const std::string& g4_model_name)
         {"UrbanMsc",            ImportModelClass::urban_msc},
         {"ICRU73QO",            ImportModelClass::icru_73_qo},
         {"WentzelVIUni",        ImportModelClass::wentzel_VI_uni},
-        {"hBrem",               ImportModelClass::h_brem},
+        {"hBrems",              ImportModelClass::h_brem},
```
Comment (sethrj) [inline app/geant-exporter/ImportProcessConverter.cc:139]: You're not going to believe this, but even though the Geant4 process class `G4hBremsstrahlung` uses the string "hBrems", the Geant4 *model* class `G4hBremsstrahlungModel` uses the string "hBrem" (singular).

Just kidding, of course you believe it.

So I think all of the *string* changes need to be reverted, but I support your suggestion of normalizing *our* usage of it -- changing the `ImportModelClass` to use `::foo_brems` instead of `::foo_brem`.

**Review by sethrj — CHANGES_REQUESTED** [review]

**Comment by sethrj** [comment]
@vrpascuzzi Reminder of this open PR... the dumb Geant4 naming is all that needs updating, I believe. 😒

**Comment by vrpascuzzi** [comment]
Thanks, @sethrj. I'm on it!

File: app/geant-exporter/ImportProcessConverter.cc (hunk 4 above)
Comment (vrpascuzzi) [reply app/geant-exporter/ImportProcessConverter.cc:139]: Yes, this is where I'm trying my best to be careful! :crab:

**Comment by vrpascuzzi** [comment]
One last thing:
```
status: Opening ROOT file
unknown file: Failure
C++ exception with description "/devel/src/celeritas-project/celeritas/src/physics/em/SeltzerBergerModel.cc:74:
celeritas: feature is not yet implemented: SeltzerBergerModel::applicability" thrown in the test body.
[  FAILED  ] ImportedProcessesTest.bremsstrahlung (3 ms)
```

originating from [SeltzerBergerModel.cc:74](https://github.com/celeritas-project/celeritas/blob/master/src/physics/em/SeltzerBergerModel.cc#L74). In particular,

```
    // TODO: these energy ranges need to be the intersection of SB table data
    // and available cross sections.
```

File: src/physics/em/SeltzerBergerModel.cc [hunk 5]
```diff
@@ -69,14 +69,13 @@ SeltzerBergerModel::SeltzerBergerModel(ModelId               id,
...
-    CELER_NOT_IMPLEMENTED("SeltzerBergerModel::applicability");
+    // TODO: Do we need to load applicabilities, e.g. lower and upper, from
+    // tables?
 
     Applicability electron_applic;
     electron_applic.particle = this->host_pointers().ids.electron;
-    electron_applic.lower    = units::MevEnergy{1};
-    electron_applic.upper    = units::MevEnergy{1e5};
+    electron_applic.lower    = units::MevEnergy{10e-4};
+    electron_applic.upper    = units::MevEnergy{10e8};
```
Comment (sethrj) [inline src/physics/em/SeltzerBergerModel.cc:78]: I think you mean `1e-4` and `1e8`? Those are the ranges that are coming from the `geant-data-cat` tool for our current standard physics set for brems.

File: test/physics/em/ImportedProcesses.test.cc [hunk 6]
```diff
@@ -196,6 +197,39 @@ TEST_F(ImportedProcessesTest, photoelectric)
...
+    ASSERT_EQ(2, all_applic.size());
+    Applicability applic = *all_applic.begin();
+
+    // Test step limits
+    for (auto mat_id : range(MaterialId{materials_->num_materials()}))
+    {
+        applic.material = mat_id;
+        auto builders   = process->step_limits(applic);
+        EXPECT_TRUE(builders[VGT::macro_xs]);
+        EXPECT_TRUE(builders[VGT::energy_loss]);
```
Comment (sethrj) [inline test/physics/em/ImportedProcesses.test.cc:229]: Can you `EXPECT_FALSE(builders[VGT::range]); ` if that's what the data has? Incidentally, do we understand why there's no range even though there's a dedx? Is it just that the range is so long (de/dx so low) that it can be neglected?

**Review by sethrj — COMMENTED** [review]
Thanks for the update, Vince! One last round I think.

File: test/physics/em/ImportedProcesses.test.cc (hunk 6 above)
Comment (vrpascuzzi) [reply test/physics/em/ImportedProcesses.test.cc:229]: I believe there should be a range, yes. So a bit confused by this.

Comment (sethrj) [reply test/physics/em/ImportedProcesses.test.cc:229]: @whokion @stognini any idea why brems doesn't have range? I remember this oddity when I was working with the data over Christmas but don't recall finding out why it is the case.

Comment (vrpascuzzi) [reply test/physics/em/ImportedProcesses.test.cc:229]: See Sec. 7.2.3, 7.5 in G4 Phys. Ref. Manual. The latter seems to convert "cut-in-range" to production thresholds in kinetic energy.

Comment (stognini) [reply test/physics/em/ImportedProcesses.test.cc:229]: I am still looking for a clear answer, but range tables are calculated from dE/dx tables, as far as I remember. Is there any chance that G4 converts E to Range on the fly because it is less expensive than lookup in this case?

Comment (amandalund) [reply test/physics/em/ImportedProcesses.test.cc:229]: Looking at `G4LossTableManager` and [section 5.6.4 of this document](http://cds.cern.ch/record/602040/files/it-2002-003.pdf?version=1/#page=48) (though it describes a now obsolete class), it sounds like de/dx tables are constructed by summing the contributions from all energy loss processes (i.e. ionization and bremsstrahlung), and then the range tables are calculated from those de/dx tables. So the ionization range tables might actually account for both ionization and bremsstrahlung?

**Review by sethrj — APPROVED** [review]

**Review by sethrj — APPROVED** [review]
